### PR TITLE
catch throwable instead of exception in case of NoClassDefFoundError

### PR DESCRIPTION
**Pull request by ever4Keny**
if any of the depending jar's are missing, an NoClassDefFoundError will be throw in the event of processing. catch throwable instead of exception.

Please do not create a Pull Request without creating an issue first. 

## What is the purpose of the change

catch throwable instead of exception in case of NoClassDefFoundError

## Brief changelog

catch throwable instead of exception
## Verifying this change
if any of the depending jar's are missing, an NoClassDefFoundError will be throw in the event of processing. catch throwable instead of exception.

Follow this checklist to help us incorporate your contribution quickly and easily:

- [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [ ] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by vongosling** [comment]
@vesense Could be fittable throw throwable here?

File: rocketmq-flume/rocketmq-flume-sink/src/main/java/org/apache/rocketmq/flume/ng/sink/RocketMQSink.java [hunk 1]
```diff
@@ -174,13 +174,13 @@ public Status process() throws EventDeliveryException {
...
 
-        } catch (Exception e) {
+        } catch (Throwable e) {
             log.error("Failed to processing event", e);
 
             if (transaction != null) {
                 try {
                     transaction.rollback();
-                } catch (Exception ex) {
+                } catch (Throwable ex) {
```
Comment (wlliqipeng) [inline rocketmq-flume/rocketmq-flume-sink/src/main/java/org/apache/rocketmq/flume/ng/sink/RocketMQSink.java]: test

**Comment by vesense** [comment]
@ever4Keny Can you create an issue for tracking problem before we merging this in?
As the PR template stated:
> Please do not create a Pull Request without creating an issue first.

**Comment by vongosling** [comment]
@vesense I would like to merge it now. @ever4Keny Thanks for your pr. Very glad to your next pr, it would be helpful if you could follow @vesense kindly suggestion.

**Comment by ever4Keny** [comment]
> @vesense I would like to merge it now. @ever4Keny Thanks for your pr. Very glad to your next pr, it would be helpful if you could follow @vesense kindly suggestion.

Sure thing! I will follow the suggestion next time!

**Pull request merged**